pallets/asset-manager/src/lib.rs

```diff
@@ -510,7 +611,7 @@ pub mod pallet {
                     | Junctions::X5(Junction::Parachain(para_id), ..)
                     | Junctions::X6(Junction::Parachain(para_id), ..)
                     | Junctions::X7(Junction::Parachain(para_id), ..)
-                    | Junctions::X8(Junction::Parachain(para_id), ..) => Some(*para_id),
+                    | Junctions::X8(Junction::Parachain(para_id), ..) => Some(para_id),
```

Comment:
We can simplify with `first_interior` like this:
```rust
            location.and_then(|location: MultiLocation| {
                if let Some(Parachain(para_id)) = location.first_interior() {
                    Some(*para_id)
                } else {
                    None
                }
            })
```

Reply:
great